### ISSUE TITLE
Sympy simplify() does not act just by having a variable multiplying some trigonometric expressions

### ISSUE THREAD

**Issue by ManuelMBULL** [sympy/sympy#17633]
Python3 version is 3.7.3-1
Sympy version is 1.3-2
Scipy version is 1.2.2-7

I show an expression which is zero after simplifying. Then I just add a "-1" and the new expression is not simplified to "-1". I also observe that it is simplified to "-1" if "b" is either removed or substituted by any number. And this should not be necessary, it should work in any case.

So there is something about multiplications and additions/substractions that confuses sympy.simplify().

Here is the "ejemplo.py" code:

```
import sympy as sp

a,b = sp.symbols( 'a,b', real = True )

x = 4 * b * sp.sin(a) ** 3 - 3 * b * sp.sin(a) + b * sp.sin(3*a)

print('Before sp.simplify           :', x   )
y = sp.simplify(x)
print('After sp.simplify            :', y   )

print('')

print('Expression -1                :', x-1 )
z = sp.simplify(x - 1)
print('Effect of -1 over sp.simplify:', z   )

print('\nIf b is removed, -1 is obtained. But it should not be necessary')
x1 = 4 * sp.sin(a) ** 3 - 3 * sp.sin(a) + sp.sin(3*a) -1
x1 = sp.simplify(x1)
print('Expected output (removing)   :', x1 )

x2 = (x-1).subs({b:823910})
x2 = sp.simplify(x2)
print('Expected output (any subs)   :', x2 )
```

Code output:

```
Before sp.simplify           : 4*b*sin(a)**3 - 3*b*sin(a) + b*sin(3*a)
After sp.simplify            : 0

Expression -1                : 4*b*sin(a)**3 - 3*b*sin(a) + b*sin(3*a) - 1
Effect of -1 over sp.simplify: 4*b*sin(a)**3 - 3*b*sin(a) + b*sin(3*a) - 1

If b is removed, -1 is obtained. But it should not be necessary
Expected output (removing)   : -1
Expected output (any subs)   : -1


------------------
(program exited with code: 0)
Press return to continue
```

**Comment by asmeurer** [comment]
fu() does the right thing. I guess it isn't used inside of simplify().

**Comment by asmeurer** [comment]
For me fu() simplifies the expression but simplify() and trigsimp() do not.

**Comment by ManuelMBULL** [comment]
> fu() is called by trigsimp().

@jksuom sympy.trigsimp() may use sympy.fu(), at least when typing "help(sympy.trigsimp)" it says "[...] Valid choices are 'matching' (default), 'groebner', 'combined', and 'fu' [...]".

> For me fu() simplifies the expression but simplify() and trigsimp() do not.

@asmeurer Maybe sympy.simplify() despite using sympy.trigsimp() only tries its default method. So for a more complete exploration, sympy.simplify() might need to explore all sympy.trigsimp() methods in a future version of Sympy.

**Comment by jksuom** [comment]
I have no problem with trigsimp.
```Python 2.7.12 (default, Aug 22 2019, 16:36:40) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import sympy as sp
>>> a,b = sp.symbols( 'a,b', real = True )
>>> x = 4 * b * sp.sin(a) ** 3 - 3 * b * sp.sin(a) + b * sp.sin(3*a)
>>> sp.trigsimp(x)
0
```

**Comment by ManuelMBULL** [comment]
> I have no problem with trigsimp.
> 
> ```
> Python 2.7.12 (default, Aug 22 2019, 16:36:40) 
> [GCC 5.4.0 20160609] on linux2
> Type "help", "copyright", "credits" or "license" for more information.
> >>> import sympy as sp
> >>> a,b = sp.symbols( 'a,b', real = True )
> >>> x = 4 * b * sp.sin(a) ** 3 - 3 * b * sp.sin(a) + b * sp.sin(3*a)
> >>> sp.trigsimp(x)
> 0
> ```

@jksuom Defining "x" as you do, just try "sp.trigsimp(x-1)", you will see it does not give "-1" but "4*b*sin(a)**3 - 3*b*sin(a) + b*sin(3*a) - 1" instead of just "-1".

**Comment by jksuom** [comment]
Hmm.. I think that trigsimp should be able to simplify also `x - 1`.

**Comment by ManuelMBULL** [comment]
> Hmm.. I think that trigsimp should be able to simplify also `x - 1`.

@jksuom I agree, it should. The problem is that it does not as I report originally. Nor sympy.simplify() nor sympy.trigsimp() do that currently. Only sympy.fu() suggested by you and @asmeurer before does it for now.

I hope this issue can help developers so sympy.simplify() and sympy.trigsimp() can do it too.

**Comment by asmeurer** [comment]
I'm still not clear why trigsimp doesn't call fu() by default. I recall there were some issues with it being too slow on some larger expressions due to calls to factor().

**Comment by jksuom** [comment]
It seems that trigsimp will also call factor() but not in exactly same way as fu().
https://github.com/sympy/sympy/blob/58e78209c8577b9890e957b624466e5beed7eb08/sympy/simplify/trigsimp.py#L1132-L1138
That will help when the constant is also a multiple of b:
```>>> import sympy as sp
>>> a,b = sp.symbols( 'a,b', real = True )
>>> x = 4 * b * sp.sin(a) ** 3 - 3 * b * sp.sin(a) + b * sp.sin(3*a)
>>> sp.trigsimp(x - b)
-b
```
In the general case, the `sin(a)**3` term is transformed by TR8, but the result is lost as nothing is done for its simplification. That could be handled by something like this:
```
--- a/sympy/simplify/trigsimp.py
+++ b/sympy/simplify/trigsimp.py
@@ -1148,7 +1148,7 @@ def _futrig(e, **kwargs):
         [identity, lambda x: _eapply(_mexpand, x, trigs)],
         TR10i,  # sin-cos products > sin-cos of sums
         TRmorrie,
-        [identity, TR8],  # sin-cos products -> sin-cos of sums
+        [identity, lambda x: _mexpand(TR8(x))],  # sin-cos products -> sin-cos of sums
         [identity, lambda x: TR2i(TR2(x))],  # tan -> sin-cos -> tan
         [
             lambda x: _eapply(expand_mul, TR5(x), trigs),
```
